src/olympia/devhub/tests/test_tasks.py

```diff
@@ -476,6 +500,111 @@ def test_count_listed_signable_addons(self):
         )
 
 
+class TestRunAddonsLinter(ValidatorTestCase):
+
+    def setUp(self):
+        super(TestRunAddonsLinter, self).setUp()
+
+        self.create_switch('webextensions')
+        self.create_switch('addons-linter')
+
+        valid_path = get_addon_file('valid_webextension.xpi')
+        invalid_path = get_addon_file('invalid_webextension_missing_id.xpi')
+
+        self.valid_upload = FileUpload.objects.create(path=valid_path)
+        self.invalid_upload = FileUpload.objects.create(path=invalid_path)
+
+    def get_upload(self, upload):
+        return FileUpload.objects.get(pk=upload.pk)
+
+    @mock.patch('olympia.devhub.tasks.run_addons_linter')
+    def test_run_linter(self, run_linter):
+        run_linter.return_value = '{"errors": 0}'
+
+        assert not self.valid_upload.valid
+
+        tasks.validate(self.valid_upload)
+
+        upload = self.get_upload(self.valid_upload)
+        assert upload.valid, upload.validation
+
+    def test_run_linter_pass(self):
+        assert not self.valid_upload.valid
+
+        tasks.validate(self.valid_upload)
+
+        upload = self.get_upload(self.valid_upload)
+        assert upload.valid, upload.validation
```

Comment:
They are similar but `test_run_linter` tests that the method get's really called, `test_run_linter_pass` assumes it's being called if that makes sense.